lib/pure/parsecsv.nim

```diff
@@ -72,12 +72,19 @@ proc raiseEInvalidCsv(filename: string, line, col: int,
                       msg: string) {.noreturn.} =
   var e: ref CsvError
   new(e)
+
+  # Prevent nil errors - these will not be set if file unable to be opened
+  let 
+    filename = if not filename.isNil(): filename else: ""
```

Comment:
Sorry, but this still isn't the way to fix this. You should ensure that ``filename`` is never ``nil`` as early as possible. In this case it is [here](https://github.com/nim-lang/Nim/pull/6149/files#diff-7d80206b6bdf8833e325c1992794b8d2R104). All this needs is a simple ``if my.filename.isNil(): my.filename = ""``

And ``msg`` should definitely never be ``nil``, silently ignoring problems like this is bad.